File: LeetCode/1431_Kids With the Greatest Number of Candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        List<Boolean> ans = new ArrayList<Boolean>();
        int max = candies[0];
        int n = candies.length;
        for(int i = 1; i<n; i++)
        {
            if(candies[i]>max)
            {
                max = candies[i];
            }
        }
        for(int i = 0; i<n; i++)
        {
            if((candies[i]+extraCandies)>=max)
                ans.add(true);
            else 
                ans.add(false);
        }
        return ans;
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> ans;
        int e = extraCandies;
        
        for(int i=0;i<candies.size();i++)
        {
            int c=0;
            for(int j=0;j<candies.size();j++)
            {
                if(candies[i]+e>=candies[j])
                    c++;
            }
            if (c==(candies.size()))
                ans.push_back(1);
            else
               ans.push_back(0);
        }
        return ans;
        
        // int max = *max_element(candies.begin(),candies.end());
        // for(int i=0;i<candies.size();i++){
        //     int a = candies[i]+extraCandies;
        //     if(a>=max){
        //         ans.push_back(1);
        //     }
        //     else
        //         ans.push_back(0);
        // }
        // return ans;
    }
};